app/models/plate/creator.rb

```diff
@@ -42,19 +47,28 @@ def execute(source_plate_barcodes, barcode_printer, scanned_user, creator_parame
   end
 
   def create_plates_from_tube_racks!(tube_racks, barcode_printer, scanned_user, _creator_parameters = nil)
+    @created_plates = []
     plate_purpose = plate_purposes.first
     plate_factories = tube_rack_to_plate_factories(tube_racks, plate_purpose)
-    return false unless plate_factories.all?(&:valid?)
+    unless plate_factories.all?(&:valid?)
+      errors = plate_factories.map(&:error_messages)
+      raise PlateCreationError, "Plate creation failed with the following errors: #{errors}"
+    end
 
     ActiveRecord::Base.transaction do
-      plate_factories.each(&:save)
+      plate_factories.each do |factory|
+        factory.save
+        add_created_plates([], factory.tube_rack, [factory.plate])
+      end
     end
-
     print_job = LabelPrinter::PrintJob.new(barcode_printer.name,
                                            LabelPrinter::Label::PlateCreator,
-                                           plates: plate_factories.map(&:plate),
+                                           plates: created_plates.pluck(:destinations).flatten.compact,
                                            plate_purpose: plate_purpose, user_login: scanned_user.login)
-    return false unless print_job.execute
+
+    unless print_job.execute
+      raise PlateCreationError, 'Barcode labels failed to print.'
+    end
```

Comment:
I added this in because I was changing the return type to be a list of plates rather than a Boolean, but since it's back to a Boolean, wonder whether we should take this exception out? I think it will prevent the plates from being created if the printing failed, which is changing the current behaviour. Rich C was just saying that the barcode printers are failing quite a lot, so it might be annoying if this prevents creation of the plates as well. What do you think?

Reply:
Agreed, especially given https://github.com/sanger/sequencescape/issues/2602

Reply:
I tried it locally and saw the message so it is working as expected.

Reply:
This is fine, the behaviour I saw was because when I checked it out, the whole method was in a transaction and therefore was rolling back the plate creation.

E's taken it out of the transaction so now behaviour will be unchanged.